pkgs/development/tools/misc/binutils/0001-libtool.m4-update-macos-version-detection-block.patch

```diff
@@ -1,47 +1,24 @@
-From 33a8dc728eb5da3e1d3439c96810d1f6b2660b89 Mon Sep 17 00:00:00 2001
-From: Andrew Childs <andrew.childs@bibo.com.ph>
-Date: Tue, 22 Feb 2022 12:24:46 +0900
+From 368f26bfece3899a8c992cfec66427266918ab80 Mon Sep 17 00:00:00 2001
+From: Randy Eckenrode <randy@largeandhighquality.com>
```

Comment:
Does it fix https://sourceware.org/PR31056? Worth mentioning it here explicitly and sending the fix upstream? Or it's already there?

Reply:
I don’t think this is the right fix for upstream. It doesn’t even try to address the versioning logic in `libtool.m4`. It’s a replacement for the previous patch, which I assume had stopped applying due to [079b20e](https://sourceware.org/git/?p=binutils-gdb.git;a=commitdiff;h=047b13da37d308778c91bfb744e8d695e56af77d;hp=079b20e89d3c511716bafb3369336e2b1c165bd5).

What’s the correct fix? I don’t know. Upstream could update the deployment target check, add a probe for `darwin2*`, or even probe the linker to see which features it supports for weak linking. Homebrew doesn’t apply any patches, so it may be that our linker is too old, or there’s something weird about our environment.